activestorage/app/models/active_storage/blob.rb

```diff
@@ -142,13 +146,14 @@ def service_headers_for_direct_upload
   #
   # Prior to uploading, we compute the checksum, which is sent to the service for transit integrity validation. If the
   # checksum does not match what the service receives, an exception will be raised. We also measure the size of the +io+
-  # and store that in +byte_size+ on the blob record.
+  # and store that in +byte_size+ on the blob record. The content type is automatically extracted from the +io+ unless
+  # the user has specified a +content_type+ and set the +identify+ as false.
   #
   # Normally, you do not have to call this method directly at all. Use the factory class methods of +build_after_upload+
   # and +create_after_upload!+.
-  def upload(io)
+  def upload(io, identify: true)
     self.checksum     = compute_checksum_in_chunks(io)
-    self.content_type = extract_content_type(io)
+    self.content_type = extract_content_type(io) if self.content_type.nil? || identify
```

Comment:
`content_type` doesn’t need an explicit receiver in the conditional. Replace `self.content_type.nil?` with `content_type.nil?`.

Reply:
Removed reference to content_type explicit receiver here 606f42457796258f9f94622678f5d4aac10b6e29